### PR TITLE
Make the CI build use the new RedHat Che distribution build...

### PR DESCRIPTION
**Pull request by davidfestal**
... available in branch
[che-176-rh-che-assembly](https://github.com/redhat-developer/rh-che/tree/che-176-rh-che-assembly)

Also related to the Jira issue: https://issues.jboss.org/browse/CHE-176

More precisely, now we create 3 docker images:
- one for the upstream Che (`openshift-conector` branch) : same as
before,
- one for the RedHat Che distribution (that integrates RH plugins such
Keycloak integration, etc ...) that also contains the Dashboard
- one for the RedHat Che distribution (that integrates RH plugins such
Keycloak integration, etc ...) but *without* the dashboard.

Signed-off-by: David Festal <dfestal@redhat.com>

### CODE REVIEW

**Comment by davidfestal** [comment]
@l0rd could you review this PR ?

File: cico_build.sh [hunk 1]
```diff
@@ -22,35 +22,60 @@ systemctl start docker
...
-  TAG=$(git rev-parse --short HEAD)
-  
-  # lets change the tag and push it to the registry
-  docker tag eclipse/che-server:nightly rhche/che-server:nightly
-  docker tag eclipse/che-server:nightly rhche/che-server:${TAG}
-  docker login -u rhchebot -p $RHCHEBOT_DOCKER_HUB_PASSWORD -e noreply@redhat.com
-  docker push rhche/che-server:nightly
-  docker push rhche/che-server:${TAG}
+  cd rh-che
+  RH_CHE_TAG=$(git rev-parse --short HEAD)
```
Comment (kbsingh) [inline cico_build.sh]: isnt this what $TAG already is ?

File: cico_build.sh [hunk 2]
```diff
@@ -22,35 +22,60 @@ systemctl start docker
...
+    bash ./build.sh
+    if [ $? -ne 0 ]; then
+      echo 'Docker Build Failed'
+      exit 2
+    fi
+    
+    # lets change the tag and push it to the registry
+    docker tag eclipse/che-server:nightly rhche/che-server:${NIGHTLY}
+    docker tag eclipse/che-server:nightly rhche/che-server:${TAG}
+    docker login -u rhchebot -p $RHCHEBOT_DOCKER_HUB_PASSWORD -e noreply@redhat.com
```
Comment (kbsingh) [inline cico_build.sh]: make sure we dont leak this in the logs etc, wrap with suiteable set +x / -x

**Comment by kbsingh** [comment]
look good to me, just check to make sure you wrap the login/pass so its not leaking - beyond that, will go with @l0rd review of the actual data flow

File: cico_build.sh (hunk 2 above)
Comment (davidfestal) [reply cico_build.sh]: well, I just moved this line, but didn't touched it afaik. So @l0rd will be able to answer your comment more than me.

**Review by l0rd — APPROVED** [review]

**Pull request merged**